src/itehandler/IteToSwitch.cc

```diff
@@ -0,0 +1,283 @@
+//
+// Created by prova on 04.08.20.
+//
+
+#include "IteToSwitch.h"
+#include "Logic.h"
+
+ite::Node::Node(PTRef term, PTRef cond, NodeRef true_node, NodeRef false_node, uint32_t id)
+    : id(id)
+    , type(cond != PTRef_Undef ? static_cast<bool>(NodeType::inner) : static_cast<bool>(NodeType::leaf))
+    , term(term)
+{
+    if (cond != PTRef_Undef) {
+        // a leaf node.
+        setCond(cond);
+        setTrueChild(true_node);
+        setFalseChild(false_node);
+    }
+}
+
+ite::NodeRef ite::Dag::newNode(PTRef val) {
+    assert(not nodes.has(val));
+    nodeRefs.push(na.alloc(val, PTRef_Undef, NodeRef_Undef, NodeRef_Undef));
+    nodes.insert(val, nodeRefs.last());
+    return nodeRefs.last();
+}
```

Comment:
It seems this could just call newNode(val, PTRef_Undef, NodeRef_Undef, NodeRef_Undef), it would avoid code duplication.